File: lab2/gr9sub2/МАЛЕЦ_ЮЛИЯ/src/fork_example.py
import os
import sys

def create_child(index):
    pid = os.fork()
    if pid == 0:
        print(f"child[{index}]: PID={os.getpid()}, PPID={os.getppid()}", flush=True)
        sys.exit(index)
    return pid

def main():
    print(f"Parent started. PID={os.getpid()}", flush=True)

    children = []
    for i in range(2):
        pid = create_child(i)
        children.append(pid)

    for _ in children:
        pid, status = os.wait()
        exit_code = os.WEXITSTATUS(status)
        print(f"Child PID={pid} exited with code {exit_code}", flush=True)

    print(f"Parent finished. PID={os.getpid()}", flush=True)

if __name__ == "__main__":
    main()

File: lab2/gr8sub2/НАСЕКО_КРИСТИНА/src/fork_example.c
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>

int main() {
    printf("parent: PID=%d\n", getpid());
    fflush(stdout);

    for (int i = 0; i < 2; i++) {
        pid_t pid = fork();
        if (pid == 0) {
            
            printf("child_%c: PID=%d, PPID=%d\n", 'A' + i, getpid(), getppid());
            fflush(stdout);
            _exit(0); 
        } else if (pid < 0) {
            perror("fork");
            return 1;
        }
    }

    
    for (int i = 0; i < 2; i++) {
        int status;
        wait(&status);
        printf("child exited with code: %d\n", WEXITSTATUS(status));
    }

    return 0;
}
